README.md
Please note that this is the development area for HelioPanel and usually does not resemble 
the most stable release. For information on installing please visit the official website 
mentioned above. HelioPanel is a control panel offered to the user's of HelioHost's 
services, which allows them to easily take full control of their website without facing any
problems. The aim of HelioPanel is to resolve any common issues that user's face every 
day, and to make website management much easier.

## Contribute ##

Any contributions to the project are welcome, including:

* Code contributions
* Requesting features
* Reporting bugs
* Improvement of wiki pages

Please follow our [Issue fixing strategy][1] when reporting issues.

## Access ##

To access HelioPanel, simply login with your existing cPanel creditentials to 
http://panel.helionet.org/.

## Updates ##
The central HelioPanel installation used in 0.3BETA and above 
(central.heliopanel.heliohost.org) is updated with the GitHub code every day at 
midnight, PST. Because of this updating feature, HelioPanel is constantly automatically 
updated with the latest features and the latest bugs solved.

## Forking Notice ##
Please be aware that the code inside this repository resembles the official central HelioPanel
installation. In order to install HelioPanel, you will need to use the official HelioPanel
project website and shouldn't download/fork the contents of the repository to install it.
However, the source code is still available if you wish to contribute to HelioPanel or 
add one of your own features!

[1]: https://github.com/HelioNetworks/HelioPanel/wiki/Issue-fixing-strategy

CHANGELOG.md
## 0.7 BETA ##
- Feature #73: Breadcrumbs on File Manager
- Feature #75: Clean up old hook.php files
- Feature #80: Improve dropdown menu
- Feature #82: Move account creation to the login page
- Bug #84: Redirects to _wdt/XXX or favicon.ico after logging in
- Bug #87: SQLBuddy redirects to http://sqlbuddy/ when not logged in
- Bug #88: Account.php redeclaration
- Bug #93: HelioPanel web directory is not allowed by default
- Bug #94: Users cannot create new accounts

## 0.6 BETA ##
- Feature #59: Remove jquery folder in favor of jquery-opimized folder
- Feature #60: Users can select account they do not own
- Feature #62: Automatically login user after account creation
- Feature #69: Reinstall invalid or old hook file
- Feature #71: Automatically submit when the account dropdown is changed
- Feature #53: Show PHP Configuration for the specified server
- Feature #67: Change the button text to something other than 'Submit Query'
- Bug #81: Fix SQLBuddy automatic login
- Bug #72: CKEditor does not work
- Bug #77: Upload file button doesn't work
- Bug #74: Update the version in the footer
- Bug #76: Correct file manager path
- Bug #81: Fix the line at the bottom of the footer

## 0.5 BETA ##
- Feature: The project has been converted to use Symfony2
- Feature #40: Store users in a database
- Feature #41: Automatically detect Ace values
- Feature #43: Link to the PHP Info page from the homepage
- Feature #46: Use an optimized version of jQuery
- Feature #47: Manage multiple accounts with one login
- Bug #44: Cannot create file or directory
- Bug #45: Cannot upload file
- Bug #49: Initial file manager path is <?php
- Bug #55: Login script doesn't work
- Bug #56: Icons are missing from file manager
- Bug #57: Ace editor is not working on hosted version
- Bug #58: Whitespace above menu bar in Google Chrome
- Bug #61: Twig runtime error
- Bug #63: Remove Symfony2 developers toolbar
- Bug #64: Files are appearing as directories
- Bug #65: SQL Buddy is showing an error
- Bug #66: Unable to save file

## 0.4 BETA ##
- Feature #30: Labelled buttons in file listing
- Feature #31: Automatically install via login.php instead of /install
- Bug #32: Back button goes to wrong directory
- Bug #33: SQLBuddy error
- Bug #36: Move renew script IFRAME to the header on every page
- Bug #39: files/*.php do not work.

## 0.3 BETA ##
- Feature #7: Ajax module boxes for file manager
- Feature #8: Automatically detect user
- Feature #9: Automatically delete install.php after installation
- Feature #21: PHP Info page
- Feature #22: Have one central HelioPanel
- Bug #25: The location of hook.php is incorrect
- Bug #26: Remove config.php
- Bug #35: Upload script doesn't work
- Bug #38: File Manager doesn't work

## 0.2 BETA ##
- Feature: SQL Buddy included
- Feature #4: Extracting .TAR and .GZ

## 0.1 BETA ##
- #1: Initial Release

header.php
<?php

session_start();

// Make sure the user is logged in
if (!isset($_SESSION['username'])) {
	header("location:login.php");
}

// Include the configuration
require 'FileRepository.php';
require 'ConfigManager.php';

$username = $_SESSION['username'];
$configManager = new ConfigManager();
$config = $configManager->getConfig();
$currentConfig = $config[$username];
$fileRepository = new FileRepository($currentConfig['hook_php'], $currentConfig['hook_auth']);

// Get the user's home directory
if (file_exists('/home/'.$username)) {
    $homedir = '/home/'.$username;
} elseif (file_exists('/home1/'.$username)) {
    $homedir = '/home1/'.$username;
} else {
	die ('Fatal Error: Cannot find home directory!');
}
?>

<html>

<head>
<link href="style.css" type="text/css" rel="stylesheet">
<link href="css/dropdown/dropdown.css" media="screen" rel="stylesheet" type="text/css" />
<link href="css/dropdown/themes/heliopanel/default.advanced.css" media="screen" rel="stylesheet" type="text/css" />
<title>HelioPanel</title>
	<script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.4/jquery.min.js"></script>
	<script type="text/javascript" src="fancybox/jquery.mousewheel-3.0.4.pack.js"></script>
	<script type="text/javascript" src="fancybox/jquery.fancybox-1.3.4.pack.js"></script>
	<link rel="stylesheet" type="text/css" href="fancybox/jquery.fancybox-1.3.4.css" media="screen" />
		<script type="text/javascript">
		$(document).ready(function() {
			$(".modal").fancybox({
				'width'				: '75%',
				'height'			: '75%',
				'autoScale'			: false,
				'type'				: 'iframe'
			});
		});
	</script>

<SCRIPT LANGUAGE="JavaScript">
<!-- Begin
function popUp(URL) {
day = new Date();
id = day.getTime();
eval("page" + id + " = window.open(URL, '" + id + "', 'toolbar=0,scrollbars=0,location=0,statusbar=0,menubar=0,resizable=0,width=400,height=200,left = 0,top = 0');");
}
// End -->
</script>

</head>

<body bgcolor="#E9C55C" style="margin-top:0px; margin-left:0px; margin-right:0px;">


<div align="center">

<table width="100%" cellspacing=0 cellpadding=5 style="border-radius:10px;" bgcolor="#FFFFFF">
<tr>

<td background="images/headerbg.png"></td>
<td background="images/header-shadow-l.png" width=5></td>

<td onclick="window.location='./'" width=714 height=102 style="background-image:url('images/toplogo.png'); background-repeat:no-repeat; cursor:pointer;">
	<p align="right"><font face="Verdana" color="#FFFFFF" size="5"><b>HelioPanel</b><br>
	</font></td>

<td background="images/header-shadow-r.png" width=5></td>
<td background="images/headerbg.png"></td>

</tr>
<tr>
<td height=5 bgcolor="#DAAE36"></td>
<td height=5 bgcolor="#DAAE36"></td>
<td width=714 height=28 bgcolor="#DAAE36">
<ul class="dropdown dropdown-horizontal">
<li><a href="http://www.heliohost.org/home/" title="HH">HelioHost</a>
	        <ul>
	            <li><a href="http://www.heliohost.org/home/">Home</a></li>
	            <li><a href="http://www.heliohost.org/home/features">Features</a>
	                <ul>
	                    <li><a href="javascript:;">Scripting Languages</a>
			            <ul>
			              <li><a href="http://www.heliohost.org/home/features/languages/php">PHP</a></li>
			              <li><a href="http://www.heliohost.org/home/features/languages/aspnet" title="Johnny Only">ASP.NET</a></li>
			              <li><a href="http://www.heliohost.org/home/features/languages/ruby" title="Johnny Only">Ruby/Ruby on Rails</a></li>
			              <li><a href="http://www.heliohost.org/home/features/languages/python">Python &amp; Django</a></li>
			              <li><a href="http://www.heliohost.org/home/features/languages/jsp" title="Johnny Only">Java/JSP</a>
			                <ul>
			                  <li><a href="http://www.helionet.org/index/index.php?showforum=74">Request Java/JSP</a></li>
			                </ul>
			              </li>
			              <li><a href="http://www.heliohost.org/home/features/languages/perl">Perl</a></li>
			              <li><a href="http://www.heliohost.org/home/features/languages/ssi">SSI</a></li>
			            </ul>
	                    </li>
	                    <li><a href="javascript:;">Database Engines</a>
			            <ul>
			              <li><a href="http://www.heliohost.org/home/features/databases/mysql">MySQL</a></li>
			              <li><a href="http://www.heliohost.org/home/features/databases/postgresql">PostgreSQL</a></li>
			              <li><a href="http://www.heliohost.org/home/features/databases/sqlite">SQLite</a></li>
			            </ul>
	                    </li>
	                    <li><a href="http://www.heliohost.org/home/features/webspace">Space and Bandwidth</a></li>
	                    <li><a href="http://www.heliohost.org/home/features/domains">URL &amp; Domains</a></li>
	                    <li><a href="http://www.heliohost.org/home/features/email">E-mail</a></li>
	                    <li><a href="http://www.heliohost.org/home/features/system">System &amp; Software</a></li>
	                </ul>
	            </li>
	            <li><a href="javascript:;">Signup</a>
	                <ul>
	                    <li><a href="http://www.heliohost.org/home/signup/stevie">Stevie</a></li>
	                    <li><a href="http://www.heliohost.org/home/signup/johnny">Johnny</a></li>
	                    <li><a href="http://www.heliohost.org/home/signup">Server Selection</a></li>
	                    <li><a href="http://heliohost.uni.cc/sign-up.php">When To Sign Up</a></li>
	                </ul>
	            </li>
	            <li><a href="javascript:;">Support</a>
		        <ul>
		          <li><a href="http://www.helionet.org/index/index.php?showforum=48">Questions</a></li>
		          <li><a href="http://www.helionet.org/index/index.php?showforum=45">Customer Service</a></li>
		        </ul>
	            </li>
	            <li><a href="http://www.heliohost.org/home/about">About</a></li>
	            <li><a href="javascript:;">Social</a>
	                <ul>
	                    <li><a href="https://www.facebook.com/pages/HelioHost/133105633434140">Facebook</a></li>
	                    <li><a href="http://twitter.com/HelioHost">Twitter</a></li>
	                </ul>
	            </li>
	            <li><a href="http://cl58services.co.cc/heliohost/data/donate.php" class="modal" title="Donate">Donate</a></li>
	            <li><a href="http://www.heliohost.org/h-terms.html" class="modal">Terms of Service</a></li>
	        </ul>
	    </li>
	    <li><a href="http://www.helionet.org/index/">HelioNet</a>
	        <ul>
	            <li><a href="javascript:;">General Discussion</a>
	        <ul>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=13">Website Management and Coding</a></li>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=8">Technology and the Internet</a></li>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=71">Philosophy, Politics, and Science</a></li>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=29">Art and Entertainment</a></li>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=5">Other Discusstion</a></li>
	        </ul>
	            </li>
	            <li><a href="javascript:;">HelioHost</a>
		        <ul>
		          <li><a href="http://www.helionet.org/index/index.php?showforum=48">Questions</a></li>
		          <li><a href="http://www.helionet.org/index/index.php?showforum=45">Customer Service</a>
		            <ul>
		              <li><a href="http://www.helionet.org/index/index.php?showforum=81">Suspended &amp; Queued Accounts</a></li>
		              <li><a href="http://www.helionet.org/index/index.php?showforum=75">Escalated Requests</a></li>
		              <li><a href="http://www.helionet.org/index/index.php?showforum=74">Java Requests</a></li>
		              <li><a href="http://www.helionet.org/index/index.php?showforum=73"> Free Domain Requests</a></li>
		              <li><a href="http://www.helionet.org/index/index.php?showforum=79">Wiki Access Requests</a></li>
		            </ul>
		          </li>
		          <li><a href="http://www.helionet.org/index/index.php?showforum=46">Customer Feedback</a></li>
		        </ul>
	            </li>
	            <li><a href="javascript:;">HelioNet</a>
	        <ul>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=1">News</a></li>
	          <li><a href="http://www.helionet.org/index/index.php?showforum=4">Contact HelioNet</a></li>
	        </ul>
	            </li>
	        </ul>
	    </li>
	    <li><a href="/heliopanel/" class="active">HelioPanel</a>
        <ul>
          <li><a href="http://heliopanel.heliohost.org">The HelioPanel Project</a></li>
          <li><a href="http://heliopanel.heliohost.org/features.php">Features</a></li>
          <li><a href="http://heliopanel.heliohost.org/feedback.php">Feedback</a>
            <ul>
              <li><a href="https://github.com/HelioNetworks/HelioPanel/issues/new">Bugs &amp; Suggestions</a></li>
              <li><a href="http://heliopanel.kayako.com/Tickets/Submit">Private Support Tickets</a></li>
            </ul>
          </li>
          <li><a href="http://heliopanel.heliohost.org/support.php">Support</a>
            <ul>
              <li><a href="https://github.com/HelioNetworks/HelioPanel/wiki/_pages">Knowledge Base</a></li>
              <li><a href="http://heliopanel.kayako.com/Tickets/Submit">Private Support Tickets</a></li>
            </ul>
          </li>
          <li><a href="http://heliopanel.heliohost.org/demo.php">Demo</a></li>
          <li><a href="https://github.com/HelioNetworks/HelioPanel">Fork on Github</a></li>
         </ul>
       </li>
	        <li><a href="javascript:;">Account</a>
	        <ul>
	            <li><a href="http://cl58services.co.cc/heliohost/data/cpanellogins.php" title="cPanel Login" class="modal">cPanel Login</a>
	      </li>
	            <li><a href="http://www.heliohost.org/scripts/renew.php" title="If your account has been suspended due to inactivity, then it may be possible to reactivate your account. Go here to renew your account." class="modal">Renew Account</a>

	            </li>
	            <li><a href="http://www.heliohost.org/scripts/status.php" title="If you have recently created a new account on HelioHost, then you can view the current status of your brand new account and look at an estimated time as for when your account will be activated." class="modal">Registration Queue Status</a>

	            </li>
	            <li><a href="http://www.heliohost.org/scripts/domain.php" title="If you require your main domain to be changed, then you can do so by going here. Please also be aware that you can add Parked and Addon Domains as a useful alternative." class="modal">Domain Change</a>

	            </li>
	            <li><a href="http://www.heliohost.org/home/support/scripts/delete" title="Permanently delete your HelioHost account.">Delete Account</a>
	            </li>
	        </ul>
	    </li>
	    <li><a href="http://www.helionet.org/index/index.php?showtopic=5055">Support FAQ</a></li>
	    <li><a href="http://wiki.heliohost.org/wiki/Main_Page">Wiki</a>
	        <ul>
	            <li><a href="http://wiki.heliohost.org/wiki/Main_Page">Wiki Main</a></li>
	            <li><a href="javascript:;">Hosting</a>
	                <ul>
	                    <li><a href="http://wiki.heliohost.org/wiki/What_is_HelioHost%3F">What is HelioHost?</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/What_Heliohost_Is_Not">What HelioHost Is Not</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/HelioNet">What is HelioNet?</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/What_Does_HelioHost_Offer">What Does HelioHost Offer?</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Signing_Up">Signing Up</a></li>
	                </ul>
	            </li>
	            <li><a href="javascript:;">Accounts</a>
	                <ul>
	                    <li><a href="http://wiki.heliohost.org/wiki/SSL">SSL</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Suspension_policy">Suspension Policy</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Moving_your_account">Moving Your Account</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/International_Countries">International Countries</a></li>
	                </ul>
	            </li>
	            <li><a href="javascript:;">cPanel &amp; Management</a>
	                <ul>
	                    <li><a href="http://wiki.heliohost.org/wiki/CPanel">cPanel</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/How_to_access_cPanel">How to access cPanel</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Parked,_Addon_and_Sub_Domains">Parked, Addon and Sub Domains</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Softaculous">Softaculous</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/MySQL_Databases">MySQL Databases</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Uploading_files_to_your_site">Uploading files to your site</a></li>
	                </ul>
	            </li>
	            <li><a href="javascript:;">Programming</a>
	                <ul>
	                    <li><a href="http://wiki.heliohost.org/wiki/ASP.NET">ASP.NET</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Site_statistics">Site Statistics</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Creating_Special_Folders_in_Roundcube">Creating Special Folders in Roundcube</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/Page_caching">Page caching</a></li>
	                    <li><a href="http://wiki.heliohost.org/wiki/MySQL">MySQL</a></li>
	                </ul>
	            </li>
	            <li><a href="http://wiki.heliohost.org/wiki/">All Pages</a></li>
	            <li><a href="http://www.helionet.org/index/index.php?showforum=79">Contribute</a></li>
	        </ul>
	    </li>
                    <li><a href="http://www.cl58services.co.cc/heliohost/">HelioHost Guide</a></li>
	    <li><a href="javascript:;">Tools</a>
	        <ul>
                    <li><a href="http://www.cl58services.co.cc/heliohost/">HelioHost Guide</a></li>
	            <li><a href="http://cl58services.co.cc/heliohost/data/status.php" class="modal" title="Server Status">Server Status</a></li>
	            <li><a href="http://cl58services.co.cc/heliohost/data/load.php" class="modal" title="Server Load">Server Load</a></li>
	            <li><a href="http://heliohost.grd.net.pl/monitor/" class="modal">Server Monitor</a></li>
	            <li><a href="http://heliohost.uni.cc/sign-up.php">Sign Up Status</a></li>
	            <li><a href="http://twitter.com/heliohost">Twitter</a></li>
	            <li><a href="http://byrondallas.heliohost.org/php/tools/dns_records.php" class="modal">DNS Lookup</a></li>
	        </ul>
           </li> </ul>
</td>
<td height=5 bgcolor="#DAAE36"></td>
<td height=5 bgcolor="#DAAE36"></td>
</tr>

<tr>
<td bgcolor="#F5F1E5"></td>
<td bgcolor="#DAAE36" background="images/body-shadow-l.png"></td>
<td bgcolor="#F5F1E5" width=714>
